### PR TITLE
[python] Use AnnData 0.10.1

### PR DESCRIPTION
**Pull request by johnkerl**
**Issue and/or context:** As in #1765. There is now a 0.10.1 which solves the problem (we were not the only ones to suffer). See also #1785 for longer-term follow-up (next year).

**Changes:**

**Notes for Reviewer:**

### CODE REVIEW

File: apis/python/setup.py [hunk 1]
```diff
@@ -269,9 +269,7 @@ def run(self):
     install_requires=[
         # Needed for Python 3.7 which anndata 0.9 doesn't support but we do
         "anndata < 0.9; python_version<'3.8'",
-        # Temporary until we align with a compatibility issue in AnnData 0.10.0.
-        # This is tracked in https://github.com/single-cell-data/TileDB-SOMA/issues/1765.
-        "anndata < 0.10.0; python_version>='3.8'",
+        "anndata; python_version>='3.8'",
```
Comment (thetorpedodog) [inline apis/python/setup.py:272]: use `anndata != 0.10.0` to avoid hitting the incompatible version

**Review by thetorpedodog — APPROVED** [review]

**Review by bkmartinjr — COMMENTED** [review]
is there a need to make the same change to the somacore dependencies?

**Comment by johnkerl** [comment]
@bkmartinjr no

**Comment by johnkerl** [comment]
Not affected AFAIK

**Comment by thetorpedodog** [comment]
> Not affected AFAIK

This is correct; we don’t play with any anndata internals in somacore, so it has no restrictions in and of itself.

**Pull request merged**